tests/pkg_openwsn/main.c

```diff
@@ -0,0 +1,313 @@
+};
+
+char *_get_name(int id)
+{
+    for (unsigned i = 0; i < ARRAY_SIZE(names); i++) {
+        if (id == names[i].id) {
+            return names[i].name;
+        }
+    }
+    return NULL;
+}
+
+static int q_cmd(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    bool queue = 1;
+
+    for (uint8_t i = 0; i < QUEUELENGTH; i++) {
+        if (openqueue_vars.queue[i].creator || openqueue_vars.queue[i].owner) {
+            queue = 0;
+            uint8_t creator = openqueue_vars.queue[i].creator;
+            uint8_t owner = openqueue_vars.queue[i].owner;
+            printf("Creator: %.9s [%d], ", _get_name(creator), creator);
+            printf("Owner: %.9s [%d]\n", _get_name(owner), owner);
+        }
+    }
+    if (queue) {
+        puts("openqueue empty");
+    }
+    return 0;
+}
+
+static int q_rmv(int argc, char **argv)
+{
+    if (argc < 2) {
+        printf("usage: %s [creator]\n", argv[0]);
+        return 1;
+    }
+
+    uint8_t creator = atoi(argv[1]);
+    if (creator == 0) {
+        printf("error: invalid input value\n");
+        return 1;
+    }
+    else {
+        printf("Removing entries created by: %.9s [%d]\n", _get_name(creator), creator);
+        openqueue_removeAllCreatedBy(creator);
+    }
+    return 0;
+}
+
+static int as_cmd(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    for (int i = 0; i < MAXACTIVESLOTS; i++) {
+        switch (schedule_vars.scheduleBuf[i].type) {
+            case CELLTYPE_TX:
+                printf("neigh: %s, slot: %03i, chan: %02i, type: TX\n", \
+                       _array_2_string(
+                           schedule_vars.scheduleBuf[i].neighbor.addr_64b,
+                           8, addr_str),
+                       schedule_vars.scheduleBuf[i].slotOffset, \
+                       schedule_vars.scheduleBuf[i].channelOffset);
+                break;
+            case CELLTYPE_RX:
+                printf("slot: %03i, chan: %02i, type: RX\n", \
+                       schedule_vars.scheduleBuf[i].slotOffset, \
+                       schedule_vars.scheduleBuf[i].channelOffset);
+                break;
+            case CELLTYPE_TXRX:
+                printf("neigh: %s, slot: %03i, chan: %02i, type: RXTX\n", \
+                       _array_2_string(
+                           schedule_vars.scheduleBuf[i].neighbor.addr_64b,
+                           8, addr_str),
+                       schedule_vars.scheduleBuf[i].slotOffset, \
+                       schedule_vars.scheduleBuf[i].channelOffset);
+                break;
+            default:
+                break;
+        }
+    }
+
+    return 0;
+}
+
+static int sc_cmd(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    /* TODO allow other prefixes via shell ?!? */
+    printf("Current tasks:%i\n", scheduler_dbg.numTasksCur);
+    printf("Max tasks:    %i\n", scheduler_dbg.numTasksMax);
+
+    return 0;
+}
+
+extern int udp_cmd(int argc, char **argv);
```

Comment:
`udp server stop` was not working for me.

Reply:
I'll remove it because in openwsn `udp_resoruces_desc` don't have an unregister function, its a linked list so this could definitively be changed. But I would rather improve on this after rebasing to a newer version on `OpenWSN` that will include a sock implementation on this side (these are upcoming changes, not yet in).

Reply:
ok, so feel free to adapt the usage message then :)